glean-core/android/src/main/java/mozilla/telemetry/glean/private/JweMetricType.kt

```diff
@@ -0,0 +1,173 @@
+        }
+
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.launch {
+            LibGleanFFI.INSTANCE.glean_jwe_set_with_compact_representation(this@JweMetricType.handle, value)
+        }
+    }
+
+    /**
+     * Build a JWE value from its elements and set to it.
+     *
+     * @param header A variable-size JWE protected header.
+     * @param key A variable-size encrypted key.
+     *      This can be an empty octet sequence.
+     * @param initVector A fixed-size, 96-bit, base64 encoded Jwe initialization vector.
+     *      If not required by the encryption algorithm, can be an empty octet sequence.
+     * @param cipherText The variable-size base64 encoded cipher text.
+     * @param authTag A fixed-size, 132-bit, base64 encoded authentication tag.
+     *      Can be an empty octet sequence.
+     */
+    fun set(header: String, key: String, initVector: String, cipherText: String, authTag: String) {
+        if (disabled) {
+            return
+        }
+
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.launch {
+            LibGleanFFI.INSTANCE.glean_jwe_set(this@JweMetricType.handle, header, key, initVector, cipherText, authTag)
+        }
+    }
+
+    /**
+     * Tests whether a value is stored for the metric for testing purposes only. This function will
+     * attempt to await the last task (if any) writing to the the metric's storage engine before
+     * returning a value.
+     *
+     * @param pingName represents the name of the ping to retrieve the metric for.
+     *                 Defaults to the first value in `sendInPings`.
+     * @return true if metric value exists, otherwise false
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    @JvmOverloads
+    fun testHasValue(pingName: String = sendInPings.first()): Boolean {
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.assertInTestingMode()
+
+        return LibGleanFFI.INSTANCE
+            .glean_jwe_test_has_value(this.handle, pingName)
+            .toBoolean()
+    }
+
+    /**
+     * Returns the stored value for testing purposes only. This function will attempt to await the
+     * last task (if any) writing to the the metric's storage engine before returning a value.
+     *
+     * @param pingName represents the name of the ping to retrieve the metric for.
+     *                 Defaults to the first value in `sendInPings`.
+     * @return value of the stored metric
+     * @throws [NullPointerException] if no value is stored
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    @JvmOverloads
+    fun testGetValue(pingName: String = sendInPings.first()): String {
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.assertInTestingMode()
+
+        if (!testHasValue(pingName)) {
+            throw NullPointerException()
+        }
+        val ptr = LibGleanFFI.INSTANCE.glean_jwe_test_get_value_as_json_string(this.handle, pingName)!!
```

Comment:
Is this is really in JSON format, that should probably be documented.

Reply:
Yes, really JSON format: https://github.com/mozilla/glean/blob/main/glean-core/src/metrics/jwe.rs#L310-L312.

The original idea in the design document was to return a custom struct (e.g. `JweData`), returning a JSON string was easier on the FFI and also more in sync with what we do for example for [custom distributions](https://github.com/mozilla/glean/blob/main/glean-core/src/metrics/custom_distribution.rs#L170).

Anyways, I'll document :)

Reply:
LGTM. This matches the approach we've taken a bunch of other places (like events, for example).